pkg/operator/ceph/disruption/clusterdisruption/add.go

```diff
@@ -57,27 +62,78 @@ func Add(mgr manager.Manager, context *controllerconfig.Context) error {
 		return err
 	}
 
+	cephClusterPredicate := predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			logger.Info("create event from ceph cluster CR")
+			return true
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			oldCluster, ok := e.ObjectOld.DeepCopyObject().(*cephv1.CephCluster)
+			if !ok {
+				return false
+			}
+			newCluster, ok := e.ObjectNew.DeepCopyObject().(*cephv1.CephCluster)
+			if !ok {
+				return false
+			}
+			return !reflect.DeepEqual(oldCluster.Spec, newCluster.Spec)
+		},
+	}
+
+	// Watch for CephClusters
+	err = c.Watch(&source.Kind{Type: &cephv1.CephCluster{}}, &handler.EnqueueRequestForObject{}, cephClusterPredicate)
```

Comment:
This call to `Watch()` is only once in the lifetime of the operator pod, right? So there is one watcher that works even if you have multiple clusters?

Reply:
Not sure if I understood the first part correctly.  This watch triggers once when the ceph cluster is created (`CreateFunc`). Next time when someone updates the cluster spec (`UpdateFunc`), the watch is triggered again.

Reply:
The operator can manage multiple Ceph clusters. Does this `Watch()` cover all namespaces or just a single namespace? For example, let's say the admin creates two CephCluster CRs in namespace A and B. We need it just to watch the namespace of the cluster CR that is being added. If the cluster CR in namespace A is updated, we don't want the watcher in namespace B to be triggered.

Reply:
Suppose we have `ClusterA in NamespaceA` and `Cluster2 in NamespaceB`.  If `ClusterA` is updated, the watcher would trigger the reconcile with correct `NamespacedName`. The reconciler then fetches the cluster based on the namespace ([here](https://github.com/rook/rook/blob/713fc80d030097c15bac07c89e50566a4c339856/pkg/operator/ceph/disruption/clusterdisruption/reconcile.go#L90)).  This should ensure correct cluster is reconciled in case of multiple clusters on different namespaces.  Let me know what you think?

Reply:
Ok, looks good that we at least have the filter later when we're processing the reconcile, so nothing should work incorrectly. But I believe the `Watch()` has the capability for a particular namespace, so we could avoid the extra processing time.

Reply:
If I'm not wrong, this is already taken care by the manager [here](https://github.com/rook/rook/blob/f86f8300f924b98e7df37207276ee40a4bd33870/pkg/operator/ceph/operator.go#L157-L171).  We tell the manager which namespaces to watch for and then add controllers (using `AddToManager`).